packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/Models/RoundTripModel.Serialization.cs

```diff
@@ -276,6 +276,339 @@ RoundTripModel IJsonModel<RoundTripModel>.Create(ref Utf8JsonReader reader, Mode
+                    foreach (var prop0 in prop.Value.EnumerateObject())
+                    {
+                        if (prop0.Value.ValueKind == JsonValueKind.Null)
+                        {
+                            dictionary.Add(prop0.Name, null);
+                        }
+                        else
+                        {
+                            dictionary.Add(prop0.Name, BinaryData.FromString(prop0.Value.GetRawText()));
+                        }
+                    }
+                    optionalRecordUnknown = dictionary;
+                    continue;
+                }
+                if (prop.NameEquals("readOnlyRequiredRecordUnknown"u8))
+                {
+                    Dictionary<string, BinaryData> dictionary = new Dictionary<string, BinaryData>();
+                    foreach (var prop0 in prop.Value.EnumerateObject())
+                    {
+                        if (prop0.Value.ValueKind == JsonValueKind.Null)
+                        {
+                            dictionary.Add(prop0.Name, null);
+                        }
+                        else
+                        {
+                            dictionary.Add(prop0.Name, BinaryData.FromString(prop0.Value.GetRawText()));
+                        }
+                    }
+                    readOnlyRequiredRecordUnknown = dictionary;
+                    continue;
+                }
+                if (prop.NameEquals("readOnlyOptionalRecordUnknown"u8))
+                {
+                    if (prop.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    Dictionary<string, BinaryData> dictionary = new Dictionary<string, BinaryData>();
+                    foreach (var prop0 in prop.Value.EnumerateObject())
+                    {
+                        if (prop0.Value.ValueKind == JsonValueKind.Null)
+                        {
+                            dictionary.Add(prop0.Name, null);
+                        }
+                        else
+                        {
+                            dictionary.Add(prop0.Name, BinaryData.FromString(prop0.Value.GetRawText()));
+                        }
+                    }
+                    readOnlyOptionalRecordUnknown = dictionary;
+                    continue;
+                }
+                if (prop.NameEquals("modelWithRequiredNullable"u8))
+                {
+                    IJsonModel<ModelWithRequiredNullableProperties> instance = (IJsonModel<ModelWithRequiredNullableProperties>)GetObjectInstance(typeof(ModelWithRequiredNullableProperties));
+                    BinaryData data = BinaryData.FromString(prop.Value.GetRawText());
+                    Utf8JsonReader modelReader = new Utf8JsonReader((ReadOnlySpan<byte>)data);
+                    modelWithRequiredNullable = instance.Create(ref modelReader, options);
```

Comment:
@m-nash - I'm wondering if we can just do 
```c#
  BinaryData data = BinaryData.FromString(prop.Value.GetRawText());
  modelWithRequiredNullable = ModelReaderWriter.Read<ModelWithRequiredNullableProperties>(data, options);
```
Instead of having to generate this `GetObjectInstance` helper method ?

Reply:
This implementation does functionally work but its much less efficient then staying at the Utf8JsonReader level.

Reply:
```c#
                    IJsonModel<ModelWithRequiredNullableProperties> instance = (IJsonModel<ModelWithRequiredNullableProperties>)GetObjectInstance(typeof(ModelWithRequiredNullableProperties));
                    modelWithRequiredNullable = instance.Create(ref reader, options);
```

Reply:
Resolving as we now will call the `DeserializeX` method for model properties.